File: src/problems/leetcode-cn/198. 打家劫舍/solution/java/Solution.java
class Solution {
    public int rob(int[] nums) {
        int length = nums.length;
        if (length == 1) return nums[0];
        int[] dp = new int[length];
        dp[0] = nums[0];
        dp[1] = max(nums[0], nums[1]);
        for (int i = 2; i < length; i++) {
            dp[i] = max(dp[i-1], dp[i-2]+nums[i]);
        }
        return dp[length-1];
    }

    int max(int a, int b) {
        return a > b ? a : b;
    }
}

File: src/problems/leetcode-cn/198. 打家劫舍/solution/rust/solution.rs
impl Solution {
    pub fn rob(nums: Vec<i32>) -> i32 {
        let length = nums.len();
        if length == 1 {
            return nums[0];
        }

        let mut dp = vec![0i32;length];
        dp[0] = nums[0];
        dp[1] = nums[0].max(nums[1]);

        for i in 2..length {
            dp[i] = dp[i-1].max(dp[i-2]+nums[i])
        }

        return dp[length-1]
    }
}